File: C#/Loon2MonoGame/LoonMonoGame-Lib/loon/action/ActionBind.cs
using loon.action.map;
using loon.canvas;
using loon.geom;

namespace loon.action
{
	public interface ActionBind
	{

		 Field2D GetField2D();

		 void SetVisible(bool v);

		 bool IsVisible();

		 int X();

		 int Y();

		 float GetX();

		 float GetY();

		 float GetScaleX();

		 float GetScaleY();

		 void SetColor(LColor color);

		 LColor GetColor();

		 void SetScale(float sx, float sy);

		 float GetRotation();

		 void SetRotation(float r);

		 float GetWidth();

		 float GetHeight();

		 float GetAlpha();

		 void SetAlpha(float alpha);

		 void SetLocation(float x, float y);

		 void SetX(float x);

		 void SetY(float y);

		 bool IsBounded();

		 bool IsContainer();

		 bool InContains(float x, float y, float w, float h);

		 RectBox GetRectBox();

		 float GetContainerWidth();

		 float GetContainerHeight();

		 ActionTween SelfAction();

		 bool isActionCompleted();
	}

}

File: Java/Loon-Lite(PureJava)/LoonLiteCore/src/loon/action/ActionBind.java
/**
 * Copyright 2008 - 2012
 * 
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not
 * use this file except in compliance with the License. You may obtain a copy of
 * the License at
 * 
 * http://www.apache.org/licenses/LICENSE-2.0
 * 
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations under
 * the License.
 * 
 * @project loon
 * @author cping
 * @email：javachenpeng@yahoo.com
 * @version 0.3.3
 */
package loon.action;

import loon.action.map.Field2D;
import loon.canvas.LColor;
import loon.geom.RectBox;

/**
 * Loon核心接口之一，实现此接口者，才可以通过action包统一操作(接口不要求全部实现 ，但只有实现了的部分，才能进行相应的动作)
 */
public interface ActionBind {

	Field2D getField2D();

	void setVisible(boolean v);

	boolean isVisible();

	int x();

	int y();

	float getX();

	float getY();

	float getScaleX();

	float getScaleY();

	void setColor(LColor color);

	LColor getColor();

	void setScale(float sx, float sy);

	float getRotation();

	void setRotation(float r);

	float getWidth();

	float getHeight();

	ActionBind setSize(float w, float h);

	float getAlpha();

	void setAlpha(float alpha);

	void setLocation(float x, float y);

	void setX(float x);

	void setY(float y);

	boolean isBounded();

	boolean isContainer();

	boolean inContains(float x, float y, float w, float h);

	RectBox getRectBox();

	float getContainerWidth();

	float getContainerHeight();

	ActionTween selfAction();

	boolean isActionCompleted();

	int getLayer();
}
